naxsi_src/naxsi_utils.c

```diff
@@ -653,25 +653,27 @@ ngx_http_wlr_finalize_hashtables(ngx_conf_t *cf, ngx_http_dummy_loc_conf_t  *dlc
     }
   }
 
-  FILE *file;
-  char fname[1024];
+  FILE *file = NULL;
+  char *fname = NULL;
   if(dlc->whitelist_file)
   {
-    size_t fname_size = dlc->whitelist_file->len;
-    if (fname_size >= sizeof(fname)) {
-      fname_size = sizeof(fname);
+    fname = calloc(1024, sizeof(char));
+    if(fname != NULL)
+    {
+      size_t fname_size = strlen((char *)dlc->whitelist_file->data);
+      if(fname_size <= 1024) {
+      strncpy(fname, (char *)dlc->whitelist_file->data, strlen((char *)dlc->whitelist_file->data));
```

Comment:
add some indentation since it is inside the `if`

Reply:
Corrected